test/unit/mdc-textfield/foundation.test.js

```diff
@@ -502,6 +481,36 @@ test('on focus makes helper text visible to the screen reader', () => {
   td.verify(helperText.showToScreenReader());
 });
 
+test('on required validation attribute change should update invalid classes', () => {
+  const {foundation, mockAdapter} = setupTest();
+  let attributeMutate;
+  td.when(mockAdapter.registerValidationAttributeChangeHandler(td.matchers.isA(Function)))
+    .thenDo((handler) => attributeMutate = handler);
+  foundation.init();
+  attributeMutate([{attributeName: 'required'}]);
+  td.verify(mockAdapter.removeClass('mdc-text-field--invalid'));
```

Comment:
Reference the constant in `cssClasses` instead of directly putting the class name inline here and in tests below